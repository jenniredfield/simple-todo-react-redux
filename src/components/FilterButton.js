import React from "react";

const FilterButton = props => {
  const { value, changeSelected, filter, text } = props;
  const isSelectedClass =
    filter === value ? "Todo-control__button--selected" : "";

  return (
    <button
      value={value}
      onClick={changeSelected}
      className={`Todo-control__button ${isSelectedClass}`}
    >
      {text}
    </button>
  );
};

export default FilterButton;
