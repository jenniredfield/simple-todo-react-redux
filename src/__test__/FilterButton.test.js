import React from "react";
import FilterButton from "../components/FilterButton";

import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("FilterButton", () => {
  it("should render a button with given props", () => {
    const filters = [
      { value: "all", text: "All" },
      { value: "completed", text: "Completed" },
      { value: "incomplete", text: "Incomplete" }
    ];

    const mockOnclick = jest.fn();

    const component = shallow(
      <FilterButton
        text={filters[1].text}
        value={filters[1].value}
        changeSelected={mockOnclick}
        key="completed"
        filter="Completed"
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("should trigger the click function", () => {
    const filters = [
      { value: "all", text: "All" },
      { value: "completed", text: "Completed" },
      { value: "incomplete", text: "Incomplete" }
    ];

    const mockOnclick = jest.fn();

    const component = shallow(
      <FilterButton
        text={filters[1].text}
        value={filters[1].value}
        changeSelected={mockOnclick}
        filter="completed"
      />
    );

    component.simulate("click");

    expect(mockOnclick).toHaveBeenCalled();
  });

  it("should have Todo-control__button class", () => {
    const filters = [
      { value: "all", text: "All" },
      { value: "completed", text: "Completed" },
      { value: "incomplete", text: "Incomplete" }
    ];

    let filter = "all";

    const mockOnclick = jest.fn();

    const wrapper = shallow(
      <FilterButton
        text={filters[1].text}
        value={filters[1].value}
        changeSelected={mockOnclick}
        filter={filter}
      />
    );

    expect(wrapper.hasClass("Todo-control__button")).toBe(true);
  });

  it("should have Todo-control__button--selected class upon change of props", () => {
    const filters = [
      { value: "all", text: "All" },
      { value: "completed", text: "Completed" },
      { value: "incomplete", text: "Incomplete" }
    ];

    const mockOnclick = jest.fn(() => {
      wrapper.setProps({ filter: "completed" });
    });

    const wrapper = shallow(
      <FilterButton
        text={filters[1].text}
        value={filters[1].value}
        changeSelected={mockOnclick}
        filter="all"
      />
    );

    expect(wrapper.hasClass("Todo-control__button--selected")).toBe(false);

    wrapper.simulate("click");

    expect(wrapper.hasClass("Todo-control__button--selected")).toBe(true);
  });

  it("should delete one of the items in list upon clickin delete", () => {
    const filters = [
      { value: "all", text: "All" },
      { value: "completed", text: "Completed" },
      { value: "incomplete", text: "Incomplete" }
    ];

    const mockOnclick = jest.fn(() => {
      wrapper.setProps({ filter: "completed" });
    });

    const wrapper = shallow(
      <FilterButton
        text={filters[1].text}
        value={filters[1].value}
        changeSelected={mockOnclick}
        filter="all"
      />
    );

    expect(wrapper.hasClass("Todo-control__button--selected")).toBe(false);

    wrapper.simulate("click");

    expect(wrapper.hasClass("Todo-control__button--selected")).toBe(true);
  });
});
