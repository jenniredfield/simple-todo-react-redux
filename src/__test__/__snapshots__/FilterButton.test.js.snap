// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton should render a button with given props 1`] = `ShallowWrapper {}`;
